CMakeLists.txt

```diff
@@ -35,7 +35,10 @@ include(cmake/cxx_standard.cmake)
 add_compile_definitions(BUILDING_ORT_GENAI_C)
 if(MSVC)
   # set updated value for __cplusplus macro instead of 199711L
-  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:/Zc:__cplusplus>)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
+
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /w15038 /wd4100 /W4 /WX")
```

Comment:
Why disable 4100?

Reply:
I don't really know. This is from the existing CMake presets.